zipkin-storage/elasticsearch-http/src/main/java/zipkin/storage/elasticsearch/http/HttpClient.java

```diff
@@ -289,23 +322,18 @@ public void clear(String index) {
     checkState(!"RED".equalsIgnoreCase(status), "Health status is RED");
   }
 
-  private static final class IndicesHealth extends GenericResultAbstractAction {
+  private static final class IndicesHealth extends Health {
```

Comment:
Was there a race condition? I only saw a janky ordering thing (with the constructor calling setURI out of order with its descendant) but it was limited to the constructing single thread (no leaking `this` outside the class that I saw, at least).

I'll swap the base class out and fix up LenientSearch (which has sorta the same problem, except it has no constructor parameters for now). In that case, though, it seems I just misunderstood the Jest API and that class can go away entirely.